Sofa/Component/LinearSolver/Direct/src/sofa/component/linearsolver/direct/SparseLUSolver.inl

```diff
@@ -54,6 +54,7 @@ void SparseLUSolver<TMatrix,TVector,TThreadManager>::solve (Matrix& M, Vector& x
 {
     SparseLUInvertData<Real> * invertData = (SparseLUInvertData<Real>*) this->getMatrixInvertData(&M);
     int n = invertData->A.n;
+    double *tmp = (double *) cs_malloc (invertData->A.n, sizeof (double)) ;
```

Comment:
Is allocating/freeing data on the heap and accessing it using raw pointer absolutly needed ? 
Can a smart pointer be used here to avoid the need for malloc/free pairs ? If it cannot, maybe there is room for a RAII design pattern.

Reply:
I cannot put the vector as an attribute because it would then not be thread safe.
I try to keep it as close as SuiteSparse as possible.